src/Components/Components/src/Binding/CascadingModelBinder.cs

```diff
@@ -136,4 +133,54 @@ void IDisposable.Dispose()
     {
         Navigation.LocationChanged -= HandleLocationChanged;
     }
+
+    bool ICascadingValueComponent.CanSupplyValue(Type valueType, string? valueName)
```

Comment:
I'm trying to make sense of whether this can interfere with unrelated `[CascadingParameter]` scenarios. It looks like it might do.

 * Right now I know you've limited this so that `FormValueSupplier.CanBind` only returns `true` for `string`. So it looks like it would interfere with any case where someone has a parameter like `[CascadingValue] public string Something { get; set; }`. That's a pretty niche scenario, but still would be very weird for a form value to show up.
 * If we want to extend in the future also to binding arbitrary types to model-bind a form post, then maybe it would start interfering with everything as it would automatically want to supply values for all types. I'm not 100% sure about this but it seems possible.

The two main mitigations I could thing of are:

1. We could require these parameters to have a special type, e.g., `[SupplyParameterFromForm] public FormData Form { get; set; }` (possibly also `FormData<T>` for model binding other types, which also gives a way to trigger binding procedurally and possibly collect validation errors)
2. Or, extend the `ICascadingValueComponent` interface to have another overload of `CanSupplyValue` which also accepts the attribute on the parameter (i.e., the `CascadingParameterAttribute` or derived attribute type), and then the logic here can be prefixed with `if (thatAttribute is not SupplyParameterFromFormAttribute) return false` and we then know for sure it's not interfering with anything else, and we're not incurring any perf cost of instantiating strings for every unrelated named `[CascadingParameter]` like on line 141.

Reply:
@SteveSandersonMS Mackinnon is working on that part. We'll very likely do 2. Which allows us to filter.

Reply:
I am scoping that part out of my PR.

Reply:
I find it difficult to review this PR in isolation because it's adding things we know we don't want, so would have to combine it with a follow-up PR to make sense of whether the net result actually adds the right things. If you feel that's definitely the best way to proceed then OK, but my preference would be to split things up into small items which are each individually correct, for example first making the tweak to generalize cascading parameters, then as a follow-up building `[SupplyParameterFromForm]` on top of that.